File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/Runtime/EmployeeRepository.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Model
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
	Public Class EmployeeRepository
		Inherits DbRepository(Of Employee, Integer, DepartmentContext)
		Implements IEmployeeRepository
		Public Sub New(ByVal unitOfWork As DbUnitOfWork(Of DepartmentContext))
			MyBase.New(unitOfWork, Function(context) context.Set(Of Employee)(), Function(x) x.EmployeeID)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/Runtime/EmployeeRepository.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Model;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    public class EmployeeRepository : DbRepository<Employee, int, DepartmentContext>, IEmployeeRepository {
        public EmployeeRepository(DbUnitOfWork<DepartmentContext> unitOfWork)
            : base(unitOfWork, context => context.Set<Employee>(), x => x.EmployeeID) {
        }
    }
}